oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsDisks.java

```diff
@@ -70,6 +70,31 @@
 
     private static final int BUFSIZE = 255;
 
+    public static void updateDiskStats(HWDiskStore diskStore) {
+        Map<String,List<Object>> vals = WmiUtil.selectObjectsFrom(null, "Win32_DiskDrive",
+                "Name,Manufacturer,Model,SerialNumber,Size,Index",
+                "WHERE SerialNumber = " + diskStore.getSerial(),DRIVE_TYPES);
+        // TODO: null checks
+        diskStore.setName((String) vals.get("Name").get(0));
+        diskStore.setModel(String.format("%s %s", vals.get("Model").get(0), vals.get("Manufacturer").get(0)).trim());
+        String index = vals.get("Index").get(0).toString();
+        diskStore.setReads(MapUtil.getOrDefault(readMap, index, 0L));
```

Comment:
These are the values that change and I don't see where you update the maps.  I think you can delete this WMI call and instead call `populateReadWriteMaps()`.  (Don't think you need `populatePartitionMaps()` ... can any of that change while OSHI is running?)

Reply:
Looked again and realised they're keyed on the disk name (diskStore.getName() gives me the index into the map, therefore)

Reply:
Actually it's not that simple. (I had the first thought at first.)

The Name in the disk is `\\.\PHYSICALDRIVE0` while the name in the stats is `0 C:` or similar.

One possibile fix is to repeat the query but only fetch the name and index.

Alternately, my research indicates that the Names are always consistently `\\.\PHYSICALDRIVEx` where x is the index.  

Alternately again, you will have the partition map attached and can parse the partition name `Disk #0, Partition #0` to get the index.  (I think `getMajor()` might also work.)

Reply:
Hmm.  I'm a little hesitant to trust either parsing the name or using the partition map as they are both subject to potential changes in the Win32 API (although that is historically rather stable).  I will repeat the query but fetch only the index, using the SerialNumber as the key since that is probably stable enough.  I can also use this to detect if the drive disappears.